@xen-orchestra/xapi/xoData.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+const assert = require('node:assert/strict')
+
+// This module provides a way to associate data with a XAPI object.
+//
+// The data is stored as a JSON object in the `other_config` field of the XAPI object, under the name `xo:${uuid.slice(0, 8})}.
+//
+// The UUID prefix must match the object UUID, because the `other_config` entries are preserved when objects are copied (e.g. VM.copy).
+
+/**
+ * Gets XO data associated with an object.
+ *
+ * @param {string} other_config - The other_config of the object.
+ * @param {string} uuid - The UUID of the object.
+ * @returns {Promise<object|undefined>} A Promise that resolves to the parsed data object, or undefined if there is no data.
+ */
+exports.extract = function extract({ other_config, uuid }) {
```

Comment:
Why not `get`?

Reply:
Because it does not fetch the data from XAPI but extracts it from the passed object.

Initially I wrote a `get` function but I changed my mind and went for the simpler approach.

IMHO this is clearer, but I can change if you want.